File: android/src/main/java/com/getcapacitor/community/classes/options/StartOptions.java
package com.getcapacitor.community.classes.options;

import static com.getcapacitor.community.BridgefyHelper.makeUUID;

import androidx.annotation.Nullable;
import com.getcapacitor.PluginCall;
import java.util.UUID;

public class StartOptions {

    @Nullable
    private UUID userID;

    @Nullable
    private String propagationProfile;

    public StartOptions(PluginCall call) {
        @Nullable
        String userID = call.getString("userID");
        this.setUserID(userID);

        @Nullable
        String propagationProfile = call.getString("propagationProfile");
        this.setPropagationProfile(propagationProfile);
    }

    private void setUserID(@Nullable String userID) {
        this.userID = makeUUID(userID);
    }

    private void setPropagationProfile(@Nullable String propagationProfile) {
        this.propagationProfile = propagationProfile;
    }

    @Nullable
    public UUID getUserID() {
        return userID;
    }

    @Nullable
    public String getPropagationProfile() {
        return propagationProfile;
    }
}

File: ios/Sources/BridgefyPlugin/Classes/Options/StartOptions.swift
import Foundation
import Capacitor

@objc public class StartOptions: NSObject {
    private var userID: UUID?
    private var propagationProfile: String?

    init(_ call: CAPPluginCall) {
        super.init()

        if let userID = call.getString("userID") {
            self.setUserID(userID)
        }

        if let propagationProfile = call.getString("propagationProfile") {
            self.setPropagationProfile(propagationProfile)
        }
    }

    func setUserID(_ userID: String?) {
        self.userID = Helper.makeUUID(userID)
    }
    func setPropagationProfile(_ propagationProfile: String?) {
        self.propagationProfile = propagationProfile
    }

    func getUserID() -> UUID? {
        return self.userID
    }
    func getPropagationProfile() -> String? {
        return self.propagationProfile
    }
}
